### ISSUE TITLE
x/build: keep test result logs available after success

### ISSUE THREAD

**Issue by dmitshur** [golang/go#34119]
Currently, the x/build testing infrastructure (both trybots and post-submit builders) generally makes test logs available under two conditions:

- while the test is running, streamed
- after the test completed, if it failed

After tests complete successfully, the test logs immediately become inaccessible.

I don't know the exact rationale for this design, but my guess is that it done to reduce computational and storage resource use. It was likely deemed logs are less useful if tests passed. /cc @bradfitz Is there more context on this?

### Feature Request

This feature request issue is to make them accessible on successful test completion for some amount of time (at least a month). Both on the [build dashboard](https://build.golang.org), and for trybot results.

The reasons for wanting this include:

- it's sometimes helpful to be able to know which tests ran and which tests were skipped, and to be able to confirm a specific test truly passed
- it's sometimes helpful to know which architectures were tested by trybots (and from that, to know which ones weren't), and having trybot result page with logs makes that much easier
- pretty much every CI system keeps test logs accessible regardless if tests succeeded or failed, so it has become a baseline expectation

I think this should be worth doing in order to improve the developer experience for people working on the Go project. Feedback from others is welcome.

/cc @golang/osp-team

**Comment by toothrot** [comment]
This sounds like it could be related to #13076 and #10313.

**Comment by bradfitz** [comment]
Agreed. Let's just keep them forever. They're small. We already keep the built artifacts after make.bash forever and they're way bigger.

History is because build.golang.org only had links for failures.

**Comment by dmitshur** [comment]
@toothrot Thanks for finding those issues. They support the decision to do this.

Both of those issues are from 2015 and talk about a subset of this problem, so I folder them into this higher level tracking issue. I think it'll be better for us to open new smaller issues to track this work as needed.

@bradfitz Thanks for context. @andybons has expressed his support for this too, so I think we're in favor of doing this.

**Comment by andybons** [comment]
Unless I’m missing something, the CL above fixes this issue. Should have marked it as Fixes instead of Updates.

**Comment by dmitshur** [comment]
I think coordinator still needs to be deployed with the change; re-opening to track that.

**Comment by dmitshur** [comment]
[CL 198197](https://golang.org/cl/198197) has made progress on this, but there's more to do to resolve this issue.

Consider a recent CL that ran trybots successfully:

https://go-review.googlesource.com/c/tools/+/204203/3#message-405874684174221493c77cbc2c783a2ac967f80e

The first Gerrit message says:

> TryBots beginning. Status page: https://farmer.golang.org/try?commit=bf0b4774

That link points to a page that says:

> TryBot result not found (already done, invalid, or not yet discovered from Gerrit). Check Gerrit for results.

The second Gerrit message says:

> TryBots are happy.

So it's not yet possible to access test logs when a trybot run completes.

Similarly, at https://build.golang.org, there are many squares with "ok", but none of them are links, so it's not possible to access test logs for successful post-submit builds.

**Comment by dmitshur** [comment]
> Similarly, at https://build.golang.org, there are many squares with "ok", but none of them are links, so it's not possible to access test logs for successful post-submit builds.

This part is resolved via [CL 340435](https://golang.org/cl/340435) now.

All that should be left in this issue is an ability to access test logs when a trybot run completes successfully.
